### PR TITLE
CASMMON-259: Create documentation for Observability Framework

### PR DESCRIPTION
**Pull request by sumithrb**
# Description

Documentation for Install and upgrade observability framework

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -77,6 +77,9 @@ Jenkinsfile.github                                       @Cray-HPE/csm-devops
 # IUF Items
 /operations/iuf/                                         @Cray-HPE/IUF
 
+#Observability Items
+/operations/observability                                @Cray-HPE/observability
```
Comment (rustydb) [inline .github/CODEOWNERS:81]: This team does not exist.
I think this should be `@Cray-HPE/cray-monitoring`

https://github.com/orgs/Cray-HPE/teams/cray-monitoring

**Review by rustydb — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (sumithrb) [reply .github/CODEOWNERS]: @rustydb sure, I have made this change

**Comment by rustydb** [comment]
I couldn't find it with a skim of this doc, but is there a place it denotes which URL to visit?

I tried:
- redbull-ncn-m001.hpc.amslabs.hpecorp.net:8081
- redbull-ncn-m001.hpc.amslabs.hpecorp.net:9441/metrics
- redbull-ncn-m001.hpc.amslabs.hpecorp.net:9090

**Comment by sumithrb** [comment]
> @rustydb The goss tests and IUF timing dashboard through the links -

https://grafana.cmn.frigg.dev.cray.com/d/Timing/iuf-timing-dashboard?orgId=1
https://grafana.cmn.frigg.dev.cray.com/d/Goss/goss-tests?orgId=1

@shreni123 can help you with the links for accessing the error and goss test dashboards in PIT.

File: operations/observability/Observability.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+
+### Prometheus metrics and Grafana dashboard
+
+After the above steps the `goss_tests` metrics are seen in prometheus when goss tests are run. Using these metrics we created the Grafana dashboard which shows us the goss tests details visually.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:258]:
```suggestion
After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics, Grafana dashboards are created to shows the Goss tests details visually.
```

File: operations/observability/Observability.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+Goss test dashboard features
+- The dashboard as dropdowns for product, suite and tests.
+- By default all products, suites and tests are selcted for overall goss tests result.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:262]:
```suggestion
- By default, all products, suites and tests are selected for overall Goss tests result.
```

File: operations/observability/Observability.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+     port: 9144
+```
+
+
+### Prometheus metrics and Grafana dashboard
+
+After the above steps the `goss_tests` metrics are seen in prometheus when goss tests are run. Using these metrics we created the Grafana dashboard which shows us the goss tests details visually.
+
+Goss test dashboard features
+- The dashboard as dropdowns for product, suite and tests.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:261]:
```suggestion
- The dashboard as dropdowns for product, suite, and tests.
```

File: operations/observability/Observability.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+- By default all products, suites and tests are selcted for overall goss tests result.
+- The overall product result, total number of the products, products passed, products failed and it's execution time is seen.
+- Similarly suite results and tests results are seen from the dashboard.
+- Node-wise test result are seen with failed nodes, passed nodes and the suite/test description.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:265]:
```suggestion
- Node-wise test result are seen with failed nodes, passed nodes, and the suite/test description.
```

File: operations/observability/Observability.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+### Prometheus metrics and Grafana dashboard
+
+After the above steps the `goss_tests` metrics are seen in prometheus when goss tests are run. Using these metrics we created the Grafana dashboard which shows us the goss tests details visually.
+
+Goss test dashboard features
+- The dashboard as dropdowns for product, suite and tests.
+- By default all products, suites and tests are selcted for overall goss tests result.
+- The overall product result, total number of the products, products passed, products failed and it's execution time is seen.
+- Similarly suite results and tests results are seen from the dashboard.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:264]:
```suggestion
- Suite results and test results are seen from the dashboard.
```

File: operations/observability/Observability.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+
+
+### Prometheus metrics and Grafana dashboard
+
+After the above steps the `goss_tests` metrics are seen in prometheus when goss tests are run. Using these metrics we created the Grafana dashboard which shows us the goss tests details visually.
+
+Goss test dashboard features
+- The dashboard as dropdowns for product, suite and tests.
+- By default all products, suites and tests are selcted for overall goss tests result.
+- The overall product result, total number of the products, products passed, products failed and it's execution time is seen.
```
Comment (nrockershousen) [inline operations/observability/Observability.md:263]:
```suggestion
- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
```

File: operations/iuf/IUF.md [hunk 8]
```diff
@@ -543,3 +544,25 @@ The following actions may be useful if errors are encountered when executing `iu
 - If the source of the error can not be determined by the previous methods, details on the underlying commands executed by an IUF stage can be found in the IUF `workflows` directory. The [Stages and Hooks](#stages-and-hooks) section
   of this document includes links to descriptions of each stage. Each of those descriptions includes an **Execution Details** section describing how to find the appropriate code in the IUF `workflows` directory to understand the
   workflow and debug the issue.
+
+## Install and Upgrade Observability Framework
+
+Install and Upgrade Observability Framework(IUATCT) creates unified consistent requirements for each product including assertions for GOSS health checks as well metrics and dashboards for health checks. Also includes a unified consistent method to automatically track Time to Install(TTI) and Time to Upgrade(TTU), as well as error and pattern count across clusters and product streams.
+
+1. Establish a consistent framework for health checks/validation, metrics and reporting for all product streams for Install/Upgrade observability.
```
[the remark this answers is not in the document]
Comment (sumithrb) [reply operations/iuf/IUF.md]: I have removed this duplicated information from the IUF.md file and added only the intro text.

**Review by rustydb — APPROVED** [review]

**Comment by rustydb** [comment]
Fixed the lint and spelling checks.

File: operations/iuf/IUF.md [hunk 9]
```diff
@@ -543,3 +544,11 @@ The following actions may be useful if errors are encountered when executing `iu
 - If the source of the error can not be determined by the previous methods, details on the underlying commands executed by an IUF stage can be found in the IUF `workflows` directory. The [Stages and Hooks](#stages-and-hooks) section
   of this document includes links to descriptions of each stage. Each of those descriptions includes an **Execution Details** section describing how to find the appropriate code in the IUF `workflows` directory to understand the
   workflow and debug the issue.
+
+## Install and Upgrade Observability Framework
+
+Install and Upgrade Observability Framework creates unified consistent requirements for each product including assertions for Goss health checks, as well metrics and dashboards for health checks. The framework also includes a unified consistent method
+to automatically track Time to Install (TTI) and Time to Upgrade (TTU), as well as error and pattern count across clusters and product streams. Automate the deployment, installation, and configuration of the Install and Upgrade Observability framework
```
Comment (mharding-hpe) [inline operations/iuf/IUF.md:551]:
```suggestion
to automatically track Time to Install (TTI) and Time to Upgrade (TTU), as well as error and pattern count across clusters and product streams. Automate the deployment, installation, and configuration of the Install and Upgrade Observability Framework
```

File: operations/iuf/IUF.md [hunk 10]
```diff
@@ -543,3 +544,11 @@ The following actions may be useful if errors are encountered when executing `iu
 - If the source of the error can not be determined by the previous methods, details on the underlying commands executed by an IUF stage can be found in the IUF `workflows` directory. The [Stages and Hooks](#stages-and-hooks) section
   of this document includes links to descriptions of each stage. Each of those descriptions includes an **Execution Details** section describing how to find the appropriate code in the IUF `workflows` directory to understand the
   workflow and debug the issue.
+
+## Install and Upgrade Observability Framework
+
+Install and Upgrade Observability Framework creates unified consistent requirements for each product including assertions for Goss health checks, as well metrics and dashboards for health checks. The framework also includes a unified consistent method
```
Comment (mharding-hpe) [inline operations/iuf/IUF.md:550]:
```suggestion
The Install and Upgrade Observability Framework creates unified consistent requirements for each product including assertions for Goss health checks, as well metrics and dashboards for health checks. The framework also includes a unified consistent method
```

**Comment by mharding-hpe** [comment]
I assume this needs a 1.4 backport as well.

**Review by mharding-hpe — APPROVED** [review]
Fixed accidentally duplicated Prometheus output on the Observability page, along with miscellaneous linting.

File: operations/iuf/IUF.md [hunk 11]
```diff
@@ -543,3 +544,11 @@ The following actions may be useful if errors are encountered when executing `iu
 - If the source of the error can not be determined by the previous methods, details on the underlying commands executed by an IUF stage can be found in the IUF `workflows` directory. The [Stages and Hooks](#stages-and-hooks) section
   of this document includes links to descriptions of each stage. Each of those descriptions includes an **Execution Details** section describing how to find the appropriate code in the IUF `workflows` directory to understand the
   workflow and debug the issue.
+
+## Install and Upgrade Observability Framework
+
+The Install and Upgrade Observability Framework creates unified consistent requirements for each product including assertions for Goss health checks, as well metrics and dashboards for health checks. The framework also includes a unified consistent method
```
Comment (deantroe) [inline operations/iuf/IUF.md:550]:
```suggestion
The Install and Upgrade Observability Framework creates unified consistent requirements for each product including assertions for Goss health checks, as well as metrics and dashboards for health checks. The framework also includes a unified consistent method
```

File: operations/iuf/IUF.md [hunk 12]
```diff
@@ -543,3 +544,11 @@ The following actions may be useful if errors are encountered when executing `iu
...
+to automatically track Time to Install (TTI) and Time to Upgrade (TTU), as well as error and pattern count across clusters and product streams. Automate the deployment, installation, and configuration of the Install and Upgrade Observability Framework
```
Comment (deantroe) [inline operations/iuf/IUF.md:551]:
```suggestion
to automatically track Time to Install (TTI) and Time to Upgrade (TTU), as well as error and pattern counts across clusters and product streams. Automate the deployment, installation, and configuration of the Install and Upgrade Observability Framework
```

Comment (deantroe) [inline operations/iuf/IUF.md:551]: The following seems like a sentence fragment.  What is it trying to say?
```
Automate the deployment, installation, and configuration of the Install and Upgrade Observability Framework
in a CSM environment.
```

File: operations/iuf/IUF.md (hunk 11 above)
Comment (deantroe) [inline operations/iuf/IUF.md:550]: I don't understand what the following means
```
... creates unified consistent requirements for each product ...
```

File: operations/observability/Observability.md [hunk 13]
```diff
@@ -0,0 +1,345 @@
+# Install and Upgrade Observability Framework
+
+The following IUF topics are discussed in the following subsections.
```
Comment (deantroe) [inline operations/observability/Observability.md:3]:
```suggestion
The following Install and Upgrade Observability Framework topics are discussed in the following subsections.
```

**Review by rambabubolla — APPROVED** [review]

**Comment by sumithrb** [comment]
@itsvijaybs I have created this new PR - https://github.com/Cray-HPE/docs-csm/pull/3288 with just one commit and no conflicts. Thank you.

**Pull request closed without merging**